pkgs/development/python-modules/magic-wormhole/default.nix

```diff
@@ -14,49 +12,40 @@
 , tqdm
 , click
 , humanize
-, ipaddress
 , txtorcon
 , nettools
-, glibc
 , glibcLocales
 , mock
 , magic-wormhole-transit-relay
+, magic-wormhole-mailbox-server
 }:
 
 buildPythonPackage rec {
   pname = "magic-wormhole";
-  version = "0.10.5";
+  version = "0.11.2";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "9558ea1f3551e535deec3462cd5c8391cb32ebb12ecd8b40b36861dbee4917ee";
+    sha256 = "01fr4bi6kc6fz9n3c4qq892inrc3nf6p2djy65yvm7xkvdxncydf";
   };
 
-  checkInputs = [ mock magic-wormhole-transit-relay ];
-  buildInputs = [ nettools glibcLocales ];
-  propagatedBuildInputs = [ spake2 pynacl six attrs twisted autobahn automat hkdf tqdm click humanize ipaddress txtorcon ];
+  buildInputs = [ glibcLocales ];
+  propagatedBuildInputs = [ spake2 pynacl six attrs twisted autobahn automat hkdf tqdm click humanize txtorcon ];
+  checkInputs = [ mock magic-wormhole-transit-relay magic-wormhole-mailbox-server ];
 
   postPatch = ''
     sed -i -e "s|'ifconfig'|'${nettools}/bin/ifconfig'|" src/wormhole/ipaddrs.py
-    sed -i -e "s|if (os.path.dirname(os.path.abspath(wormhole))|if not os.path.abspath(wormhole).startswith('/nix/store') and (os.path.dirname(os.path.abspath(wormhole))|" src/wormhole/test/test_cli.py
-
-    # magic-wormhole will attempt to find all available locales by running
-    # 'locale -a'.  If we're building on Linux, then this may result in us
-    # running the system's locale binary instead of the one from Nix, so let's
-    # ensure we patch this.
-    sed -i -e 's|getProcessOutputAndValue("locale"|getProcessOutputAndValue("${glibc}/bin/locale"|' src/wormhole/test/test_cli.py
```

Comment:
I think we still need this as the point `running the system's locale binary instead of the one from Nix` should still stand right?

But a patch with `substituteAll` would be better.

Reply:
not anymore we pass `glibcLocales` as buildInput and patch to `getProcessOutputAndValue` os.environ to honor it. I've checked that the tests affected are executed correctly

Reply:
Thanks :+1: